### PR TITLE
[FIXED JENKINS-41637] Users with discover permission get broken UI when renaming job to inaccessible one

### PR DESCRIPTION
**Pull request by ksenia-nenasheva**
# Description

See [JENKINS-41637](https://issues.jenkins-ci.org/browse/JENKINS-41637).

Details:
A user with permission to configure Job A and discover Job B gets a broken UI when attempting to rename "A" to "B":

- Added checks that job with same name exists. The logic tries the current user authentication first and then falls back to SYSTEM.
- Added new message when the job exists and user has no permission to read this job.

### Changelog entries

Proposed changelog entries:

* Entry 1: Improve diagnostics of hidden existing items when renaming jobs
* ...

### Submitter checklist

- [x] JIRA issue is well described
- [x] Link to JIRA ticket in description, if appropriate
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For new API and extension points: Link to the reference implementation in open-source (or example in Javadoc)

### Desired reviewers

@mention

### CODE REVIEW

**Comment by daniel-beck** [comment]
> Added new message when the job exists and user has no permission to read this job

As the user has the permission to *Discover* that other job, the message could be more explicit about its existence, I think. "Name not valid" seems like it could cause confusion.

**Comment by ksenia-nenasheva** [comment]
Done.
User has permissions to _Read_ or _Discover_ => Message: _"The name is already in use"_.
User hasn't permissions => Message: _"The name is not valid"_.

**Comment by KostyaSha** [comment]
Please make method name consistent with variable and UI, occupied == inUse?

**Comment by daniel-beck** [comment]
Agree with @kostyasha here, naming as 'occupied' seems unusual and is inconsistent.

**Comment by ksenia-nenasheva** [comment]
@KostyaSha @daniel-beck Done.

**Comment by ksenia-nenasheva** [comment]
@jenkinsci/code-reviewers

File: core/src/main/java/hudson/model/Job.java [hunk 1]
```diff
@@ -1604,4 +1607,50 @@ public BuildTimelineWidget getTimeline() {
...
+    @Restricted(NoExternalUse.class)
+    public Boolean checkIfNameIsUsed(@Nonnull String newName) throws Failure{
+        
+        Item item = null;
+        Jenkins.checkGoodName(newName);
+        
+        try {
+            item = getParent().getItem(newName);
+        } catch(AccessDeniedException ex) {    
+            // User has Discover permissions for existing job with the same name
```
Comment (KostyaSha) [inline core/src/main/java/hudson/model/Job.java]: Or move this information into log trace level

**Comment by ksenia-nenasheva** [comment]
@KostyaSha Done.

File: core/src/main/java/hudson/model/Job.java (hunk 1 above)
Comment (KostyaSha) [inline core/src/main/java/hudson/model/Job.java]: Then you don't need comment.

Comment (ksenia-nenasheva) [inline core/src/main/java/hudson/model/Job.java]: Done

**Review by daniel-beck — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
It is not going to make things worse for sure, so 👍 .

**Pull request merged**